c7n/resources/ecs.py

```diff
@@ -525,3 +538,145 @@ def process_instance(self, client, cluster, instance):
         except (client.exceptions.NoUpdateAvailableException,
                 client.exceptions.UpdateInProgressException):
             return
+
+
+@ECSCluster.action_registry.register('tag')
+@TaskDefinition.action_registry.register('tag')
+@Service.action_registry.register('tag')
+@Task.action_registry.register('tag')
+@ContainerInstance.action_registry.register('tag')
+class TagEcsResource(Tag):
+    """Action to create tag(s) on an ECS resource
+    (ecs, ecs-task-definition, ecs-service, ecs-task, ecs-container-instance)
+
+    Requires arns in new format for tasks, services, and container-instances.
+    https://docs.aws.amazon.com/AmazonECS/latest/userguide/ecs-resource-ids.html
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: tag-ecs-service
+                resource: ecs-service
+                filters:
+                  - "tag:target-tag": absent
+                  - type: taggable
+                actions:
+                  - type: tag
+                    key: target-tag
+                    value: target-value
+    """
+    permissions = ('ecs:TagResource',)
```

Comment:
you'll also need to add a value for batch_size, in this case it should be equal to 1 see discussion here: https://github.com/cloud-custodian/cloud-custodian/pull/3568

Reply:
added